src/libraries/System.Net.HttpListener/src/System/Net/Managed/WebSockets/HttpWebSocket.Managed.cs

```diff
@@ -59,6 +60,7 @@ internal static partial class HttpWebSocket
 
             WebSocket webSocket = WebSocket.CreateFromStream(context.Connection.ConnectedStream, isServer:true, subProtocol, keepAliveInterval);
 
+            Debug.Assert(!OperatingSystem.IsBrowser());
```

Comment:
Seems not supposed to be called in browser context, asserted for now

Reply:
This is correct `WebSocket.CreateFromStream` there is no way to create a websocket from the stream level as it is not supported by native browser interface.